### PR TITLE
test 'query tip'

### PR DESCRIPTION
**Pull request by saratomaz**

### CODE REVIEW

**Comment by saratomaz** [comment]
https://github.com/input-output-hk/cardano-node/issues/3878

File: cardano_node_tests/tests/test_cli.py [hunk 1]
```diff
@@ -207,6 +207,57 @@ def test_tx_view(self, cluster: clusterlib.ClusterLib):
...
+            ).stdout.decode("utf-8")
+        )
+
+        expected_out = {
+            "block",
+            "epoch",
+            "era",
+            "hash",
+            "slot",
+            "slotInEpoch",
```
Comment (mkoura) [inline cardano_node_tests/tests/test_cli.py:230]: This will not work in latest released node version.

File: cardano_node_tests/tests/test_cli.py [hunk 2]
```diff
@@ -207,6 +207,57 @@ def test_tx_view(self, cluster: clusterlib.ClusterLib):
...
+            "slotsToEpochEnd",
+            "syncProgress",
+        }
+
+        # check that 'query tip' is returning the expected fields
+        assert (
+            set(tip_out.keys()) == expected_out
+        ), "The output fields of 'query tip' are not the expected"
```
Comment (mkoura) [inline cardano_node_tests/tests/test_cli.py:238]: I don't think that "are not the expected" is grammatically correct.
Could be "... not the expected ones", or even better "Unexpected fields in `query tip` output"

File: cardano_node_tests/tests/test_cli.py [hunk 3]
```diff
@@ -207,6 +207,57 @@ def test_tx_view(self, cluster: clusterlib.ClusterLib):
...
+
+        # check that 'slotInEpoch' is never greater than epoch length
+        assert (
+            tip_out["slotInEpoch"] <= cluster.epoch_length
+        ), "'slotInEpoch' is greater than epoch length"
+
+        # check that 'slotsToEpochEnd' is the difference between epoch length and 'slotInEpoch'
+        assert (
+            tip_out["slotsToEpochEnd"] <= cluster.epoch_length - tip_out["slotInEpoch"]
+        ), "'slotsToEpochEnd' is not the expected"
```
Comment (mkoura) [inline cardano_node_tests/tests/test_cli.py:248]: Maybe "'slotsToEpochEnd' doesn't have the expected value"?

File: cardano_node_tests/tests/test_cli.py [hunk 4]
```diff
@@ -207,6 +207,57 @@ def test_tx_view(self, cluster: clusterlib.ClusterLib):
...
+        # check that 'slot' is never greater than the total number of slots
+        assert (
+            tip_out["slot"] <= tip_out["epoch"] * cluster.epoch_length
+        ), "'slot' is greater than total number of slots"
+
+        # check that 'era' is the expected
+        expected_era = VERSIONS.cluster_era_name.title()
+        assert (
+            tip_out["era"] == expected_era
+        ), f"The 'era' reported is not the expected: {tip_out['era']} != {expected_era}"
```
Comment (mkoura) [inline cardano_node_tests/tests/test_cli.py:259]: Maybe "'era' doesn't have the expected value: ..."?

**Review by mkoura — CHANGES_REQUESTED** [review]

**Review by mkoura — APPROVED** [review]

**Pull request merged**